### PR TITLE
Feat/staked gho sm

### PR DESCRIPTION
**Pull request by foodaka**
Initial gho sm

### CODE REVIEW

File: packages/contract-helpers/src/v3-staking-contract/index.ts [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+
+    this.erc20Service = new ERC20Service(provider);
+
+    this.erc20_2612Service = new ERC20_2612Service(provider);
+
+    this.stakingContractAddress = stakingServiceConfig.TOKEN_STAKING_ADDRESS;
+  }
+
+  @SignStakingValidator
+  public async signStaking(
```
Comment (grothem) [inline packages/contract-helpers/src/v3-staking-contract/index.ts:95]: We have a method in the interface for generating the signature that you could use, should be the same as this.

`generateSignatureRequest` in `poolSlice`

Comment (foodaka) [reply packages/contract-helpers/src/v3-staking-contract/index.ts:85]: i think its being used in the stakeslice no?

Comment (grothem) [reply packages/contract-helpers/src/v3-staking-contract/index.ts:85]: ah yeah good call, nvm

File: packages/contract-helpers/src/V3-uiStakeDataProvider-contract/typechain/factories/index.ts
Comment (grothem) [inline packages/contract-helpers/src/V3-uiStakeDataProvider-contract/typechain/factories/index.ts:1]: I think this file can get deleted

File: packages/contract-helpers/src/V3-uiStakeDataProvider-contract/typechain/index.ts
Comment (grothem) [inline packages/contract-helpers/src/V3-uiStakeDataProvider-contract/typechain/index.ts:1]: this file can get deleted

File: jest.config.js [hunk 2]
```diff
@@ -41,6 +41,7 @@ module.exports = {
     'packages/contract-helpers/src/v3-migration-contract/typechain',
     'packages/contract-helpers/src/index.ts',
     'packages/math-utils/src/formatters/reserve/index.ts', // TODO: remove
+    // 'packages/contract-helpers/src/V3-uiStakeDataProvider-contract/index.ts', // TODO REMOVE, TEMP
```
Comment (grothem) [inline jest.config.js:44]: can this get removed?

File: packages/contract-helpers/src/v3-staking-contract/typechain/factories/index.ts
Comment (grothem) [inline packages/contract-helpers/src/v3-staking-contract/typechain/factories/index.ts:1]: can get deleted

File: packages/contract-helpers/src/v3-staking-contract/typechain/index.ts
Comment (grothem) [inline packages/contract-helpers/src/v3-staking-contract/typechain/index.ts:1]: can get deleted

File: packages/contract-helpers/src/V3-uiStakeDataProvider-contract/typechain/common.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
```
Comment (grothem) [inline packages/contract-helpers/src/V3-uiStakeDataProvider-contract/typechain/common.ts:2]: this tslint comment can get deleted

Comment (foodaka) [reply packages/contract-helpers/src/V3-uiStakeDataProvider-contract/typechain/common.ts]: added in 4d726e61197da749037f3ae338953449ddfc6935

File: packages/contract-helpers/src/v3-staking-contract/typechain/index.ts
Comment (foodaka) [reply packages/contract-helpers/src/v3-staking-contract/typechain/index.ts:1]: added in 4d726e61197da749037f3ae338953449ddfc6935

File: jest.config.js (hunk 2 above)
Comment (foodaka) [reply jest.config.js]: added in 4d726e61197da749037f3ae338953449ddfc6935

File: packages/contract-helpers/src/V3-uiStakeDataProvider-contract/typechain/index.ts
Comment (foodaka) [reply packages/contract-helpers/src/V3-uiStakeDataProvider-contract/typechain/index.ts:1]: deleted in 4d726e61197da749037f3ae338953449ddfc6935

**Review by grothem — APPROVED** [review]

**Pull request merged**